### PR TITLE
[CBRD-23917] Server side log processing - DML, DCL, Timer

### PR DESCRIPTION
**Pull request by joohok**
http://jira.cubrid.org/browse/CBRD-23917

This PR is for, 
1. Traversing Log records 
2. Pick interpretable log record
3. Get information which is required to make kinds of log item (heap record for DML, time info for TIMER, ..)
4. Make log info (prior stage of log item, this log info is used to generate log item at client side)

generating DDL log info will be added soon. 
and queue mechanism and thread handling will be covered in next PR.

### CODE REVIEW

File: src/transaction/log_impl.h [hunk 1]
```diff
@@ -759,6 +760,85 @@ typedef struct log_logging_stat
   unsigned long async_commit_request_count;
 } LOG_LOGGING_STAT;
 
+/*for CDC interface */
```
Comment (hornetmj) [inline src/transaction/log_impl.h:768]: Could you please separate the structures, defines, etc related to the CDC operation?

Comment (hornetmj) [reply src/transaction/log_impl.h:768]: Additionally, I think you should use/make the prefix for CDC. It makes it easy for other developers to understand your code and to sustain your code easily.

File: src/transaction/log_impl.h [hunk 2]
```diff
@@ -759,6 +760,85 @@ typedef struct log_logging_stat
...
+typedef enum dcl_type DCL_TYPE;
+typedef enum dml_type DML_TYPE;
+
+extern LOG_READER_INFO log_Reader_info;
+
+extern char *log_Infos;
+extern int log_Infos_length;
+
+/* *INDENT-OFF* */
+extern lockfree::circular_queue < LOG_INFO_ENTRY* > *log_info_queue;
```
Comment (hornetmj) [inline src/transaction/log_impl.h]: follows the coding convention when you use the global variables.

File: src/transaction/log_manager.c [hunk 3]
```diff
@@ -170,6 +170,33 @@ struct archive_log_header_scan_context
   LOG_ARV_HEADER header;
 };
 
+/*It will be moved to the CDC module and will define overflow.h in that module  */
+
+struct overflow_first_part
```
Comment (hornetmj) [inline src/transaction/log_manager.c:175]: It looks good to me if you move this structure and typedef to the CDC header file. I mean I wish you make a CDC header(s) and use it for declarations.

File: src/transaction/log_manager.c [hunk 4]
```diff
@@ -306,6 +333,26 @@ static void log_sysop_do_postpone (THREAD_ENTRY * thread_p, LOG_TDES * tdes, LOG
 
 static int logtb_tran_update_stats_online_index_rb (THREAD_ENTRY * thread_p, void *data, void *args);
 
+/*for CDC */
+static int log_reader_initialize ();
```
Comment (hornetmj) [inline src/transaction/log_manager.c]: I can't find the implementation of this declaration (this function).

**Review by hornetmj — COMMENTED** [review]
Draft-PR, but I leave some comments. Please check it.

File: src/transaction/log_impl.h (hunk 1 above)
Comment (joohok) [reply src/transaction/log_impl.h:768]: OK, I separated them, and modified names of data structures and  functions for more readability.

**Pull request closed without merging**